### PR TITLE
98 create snackbar messages

### PR DESCRIPTION
**Pull request by juli-txt**
## Description

- Added snackbar messages
- Added network status service
- Added example implementation of snackbar usage when internet connection is lost or regained 

## Type of changes

- [ ] Bugfix 
- [x] Feature 
- [ ] Refactoring 
- [ ] Optimization 
- [ ] Documentation 
- [ ] CI/CD 
- [ ] Other

### CODE REVIEW

File: src/common/services/SnackbarProvider/SnackbarProvider.hooks.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ * @category Services
+ */
+export const useSnackbarProvider = (): SnackbarProviderHookReturn => {
+  const [snackbarsErrorWarning, setSnackbarsErrorWarning] = useState<SnackbarMessageProps[]>([])
+  const [snackbarsSuccessInfo, setSnackbarsSuccessInfo] = useState<SnackbarMessageProps[]>([])
+
+  // Adds a snackbar to the array of snackbars. It depends on the severity type which array is used.
+  const addSnackbar = useCallback(
+    (newSnackbar: SnackbarMessageProps) => {
+      if (newSnackbar.severity === 'error' || newSnackbar.severity === 'warning') {
```
Comment (DimitriB01) [inline src/common/services/SnackbarProvider/SnackbarProvider.hooks.tsx:37]: const { severity, message } = newSnackbar 
macht das anschließende lesen einfacher mmn

File: src/common/services/SnackbarProvider/SnackbarProvider.hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        }
+        const rest = snackbarsSuccessInfo.length < 3 ? snackbarsSuccessInfo : snackbarsSuccessInfo.slice(0, -1)
+        setSnackbarsSuccessInfo([newSnackbar, ...rest])
+      }
+    },
+    [snackbarsErrorWarning, snackbarsSuccessInfo]
+  )
+
+  // Updates a snackbar in the array of snackbars. The message is used as a key to change an existing snackbar.
+  const updateSnackbar = useCallback((snackbarToUpdate: SnackbarMessageProps) => {
```
Comment (DimitriB01) [inline src/common/services/SnackbarProvider/SnackbarProvider.hooks.tsx:55]: das selbe wie oben

File: src/common/services/SnackbarProvider/SnackbarProvider.hooks.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    } else if (snackbarToUpdate.severity === 'success' || snackbarToUpdate.severity === 'info') {
+      setSnackbarsSuccessInfo((previous) =>
+        previous.map((snackbar) => (snackbar.message === snackbarToUpdate.message ? snackbarToUpdate : snackbar))
+      )
+    }
+  }, [])
+
+  // Removes a snackbar from the array of snackbars.
+  const removeSnackbar = useCallback((snackbarToRemove: SnackbarMessageProps) => {
+    if (snackbarToRemove.severity === 'error' || snackbarToRemove.severity === 'warning') {
```
Comment (DimitriB01) [inline src/common/services/SnackbarProvider/SnackbarProvider.hooks.tsx:69]: same

File: src/components/SnackbarContainer/SnackbarContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+ * It can't be used as a standalone component and must be somewhere beneath the snackbar provider.
+ * @returns {JSX.Element} - The snackbar container component.
+ * @category Components
+ */
+const SnackbarContainer = () => {
+  const { snackbarsErrorWarning, snackbarsSuccessInfo, addSnackbar, updateSnackbar } = useContext(SnackbarContext)
+
+  const { t } = useTranslation()
+  const isOnline = useNetworkStatus()
+  const [recentlyOffline, setRecentlyOffline] = useState(false)
```
Comment (DimitriB01) [inline src/components/SnackbarContainer/SnackbarContainer.tsx:18]: useState<boolean>(false)

File: src/components/SnackbarMessage/SnackbarMessage.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+/**
+ * SnackbarMessage presents an alert to display messages with different severities to inform users.
+ * It can't be used as a standalone component and needs to be placed inside a snackbar component.
+ * @param props - Props containing message, severity and autoHideDuration of a snackbar.
+ * @returns {JSX.Element} - The message component of a snackbar.
+ * @category Components
+ */
+const SnackbarMessage = (props: SnackbarMessageProps) => {
+  const { t } = useTranslation()
+  const [openTransition, setOpenTransition] = useState(true)
```
Comment (DimitriB01) [inline src/components/SnackbarMessage/SnackbarMessage.tsx:37]: useState (true) mit mit <> boolean

File: src/components/SnackbarContainer/SnackbarContainer.tsx (hunk 4 above)
Comment (DimitriB01) [reply src/components/SnackbarContainer/SnackbarContainer.tsx:18]: mit <> boolean

File: src/pages/App/App.tsx [hunk 6]
```diff
@@ -20,18 +20,21 @@ logBuffer()
...
-            <Route path="/dashboard" element={<Dashboard />} />
+      <SnackbarProvider>
+        <Router>
+          <Routes>
+            <Route element={<MainFrame />}>
+              <Route index element={<Home />} />
+              <Route path="/theme" element={<ThemePresentation />} />
+              <Route path="/login" element={<Login />} />
+              <Route path="/dashboard" element={<Dashboard />} />
+              <Route path="/🥚" element={<div>Ei</div>} />
```
Comment (DimitriB01) [inline src/pages/App/App.tsx:31]: Was für Ei

Comment (theUpsider) [reply src/pages/App/App.tsx:31]: Easter egg bruder

**Review by theUpsider — CHANGES_REQUESTED** [review]
Stuff that was discussed

**Review by theUpsider — APPROVED** [review]

**Pull request merged**